sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/partition_client.hpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#pragma once
+#include "models/event_data.hpp"
+#include "models/partition_client_models.hpp"
+
+#include <azure/core/amqp.hpp>
+#include <azure/core/datetime.hpp>
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/nullable.hpp>
+namespace Azure { namespace Messaging { namespace EventHubs {
+  /**brief PartitionClientOptions provides options for the CreatePartitionClient function.
+   */
+  struct PartitionClientOptions
+  {
+    /**@brief StartPosition is the position we will start receiving events from,
+     * either an offset (inclusive) with Offset, or receiving events received
+     * after a specific time using EnqueuedTime.
+     *
+     *@remark NOTE: you can also use the [Processor], which will automatically manage the start
+     * value using a [CheckpointStore]. See [example_consuming_with_checkpoints_test.go] for an
+     * example.
+     */
+    Models::StartPosition StartPosition;
+
+    /**@brief OwnerLevel is the priority for this partition client, also known as the 'epoch' level.
+     * When used, a partition client with a higher OwnerLevel will take ownership of a partition
+     * from partition clients with a lower OwnerLevel.
+     * Default is off.
+     */
+    int64_t OwnerLevel;
+
+    /**@brief Prefetch represents the size of the internal prefetch buffer. When set,
+     * this client will attempt to always maintain an internal cache of events of
+     * this size, asynchronously, increasing the odds that ReceiveEvents() will use
+     * a locally stored cache of events, rather than having to wait for events to
+     * arrive from the network.
+     *
+     * Defaults to 300 events if Prefetch == 0.
+     * Disabled if Prefetch < 0.
+     */
+
+    int32_t Prefetch = 300;
+  };
+
+  /**@brief ReceiveEventsOptions contains optional parameters for the ReceiveEvents function
+   */
+  struct ReceiveEventsOptions
+  {
+    // For future expansion
+  };
+
+  /** PartitionClient is used to receive events from an Event Hub partition.
+   *
+   * This type is instantiated from the [ConsumerClient] type, using
+   * [ConsumerClient.CreatePartitionClient].
+   */
+  class PartitionClient {
```

Comment:
BTW, desn't clang-format move the curly brace to the next line? Is it not running in CI for Eventhubs?

Reply:
It's running in the CI - apparently it doesn't always move the curly brace to the next line, but I don't knowwhat the rules are.